Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool/Config.hs

```diff
@@ -14,69 +14,30 @@
-  { minQuotesToQualifyForIntelligentPool :: Int,
-    minQuotesToQualifyForIntelligentPoolWindowOption :: SWC.SlidingWindowOptions
-  }
-  deriving (Generic, FromDhall)
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common (logInfo)
+import Kernel.Utils.Error
+import Storage.CachedQueries.CacheConfig (CacheFlow)
+import qualified Storage.CachedQueries.Merchant.DriverPoolConfig as CDP
 
 data CancellationScoreRelatedConfig = CancellationScoreRelatedConfig
   { popupDelayToAddAsPenalty :: Maybe Seconds,
     thresholdCancellationScore :: Maybe Int,
-    thresholdRidesCount :: Maybe Int
-  }
-  deriving (Generic, FromDhall)
-
-data DriverPoolConfig = DriverPoolConfig
-  { minRadiusOfSearch :: Meters,
-    maxRadiusOfSearch :: Meters,
-    radiusStepSize :: Meters,
-    driverPositionInfoExpiry :: Maybe Seconds,
-    actualDistanceThreshold :: Maybe Meters,
-    maxDriverQuotesRequired :: Int,
-    driverQuoteLimit :: Int,
-    intelligentPoolPercentage :: Maybe Int,
-    driverRequestCountLimit :: Int
-  }
-  deriving (Generic, FromDhall)
-
-data OverrideDriverPoolConfigRange = OverrideDriverPoolConfigRange
-  { startDistance :: Meters,
-    endDistance :: Maybe Meters
+    minRidesForCancellationScore :: Maybe Int
   }
-  deriving (Generic, FromDhall)
-
-data OverrideDriverPoolConfig = OverrideDriverPoolConfig
-  { configRange :: OverrideDriverPoolConfigRange,
-    driverPoolCfg :: DriverPoolConfig
-  }
-  deriving (Generic, FromDhall)
-
-type HasDriverPoolConfig r =
-  ( HasField "driverPoolCfg" r DriverPoolConfig,
-    HasField "overrideDriverPoolConfig" r [OverrideDriverPoolConfig],
-    HasField "intelligentPoolConfig" r IntelligentPoolConfig,
-    HasField "cancellationScoreRelatedConfig" r CancellationScoreRelatedConfig,
-    HasField "defaultPopupDelay" r Seconds,
-    HasMaxParallelSearchRequests r
-  )
-
-type HasMaxParallelSearchRequests r =
-  ( HasField "maxParallelSearchRequests" r Int
-  )
-
-getDriverPoolConfig :: (MonadFlow m, MonadReader r m, HasDriverPoolConfig r) => Meters -> m DriverPoolConfig
-getDriverPoolConfig dist = do
-  defaultConfig <- asks (.driverPoolCfg)
-  overrideConfigs <- asks (.overrideDriverPoolConfig)
-  let applicableConfig = find filterByDist overrideConfigs
-  return $ maybe defaultConfig (.driverPoolCfg) applicableConfig
+  deriving (Generic)
+
+getDriverPoolConfig :: (MonadFlow m, MonadReader r m, CacheFlow m r, EsqDBFlow m r) => Id Merchant -> Meters -> m DriverPoolConfig
+getDriverPoolConfig merchantId dist = do
+  configs <- CDP.findAllByMerchantId merchantId
+  let applicableConfig = find filterByDist configs
+  case configs of
+    [] -> throwError $ InvalidRequest "DriverPoolConfig not found"
+    (config : _) -> pure $ maybe config identity applicableConfig
   where
-    filterByDist cfg =
-      (dist >= cfg.configRange.startDistance)
-        && maybe True (dist <=) cfg.configRange.endDistance
+    filterByDist cfg = dist >= cfg.tripDistance
```

Comment:
did you check it. If we have trip distance config 0, 10 then for 15 it will return 0 wala config not 10 one.

Reply:
It is order by `tripDistance desc` in query so it will return 10's config